### ISSUE TITLE
what is the problem when i test on kitti ?

### ISSUE THREAD

**Issue by slz929** [hlwang1124/SNE-RoadSeg#17]
I use the example script for kitti dataset test. 
Depth image is from the kitti stereo 2015 dataset ground truth (disparity image not the depth) and corresponding to the rgb image.
I set the camera parameters unchanged, and the depth caculation from disparity as below:
``` 
        depth_image= depth_image.astype(np.float32)
        row= depth_image.shape[0]
        col= depth_image.shape[1]
        depth_image= np.true_divide(np.ones((row,col)), depth_image)
        depth_image= depth_image * 721.5377* 0.537 * 256.0 

        normal = sne_model(torch.tensor(depth_image.astype(np.float32)), camParam
```

**This is the example depth:**
![image](https://user-images.githubusercontent.com/17669514/99940750-c2726f80-2da7-11eb-86e1-72bc08216c69.png)
**The example result:**
![image](https://user-images.githubusercontent.com/17669514/99940815-e7ff7900-2da7-11eb-84b9-f2fdc61e0954.png)

**My disparity image:**
![image](https://user-images.githubusercontent.com/17669514/99941008-40367b00-2da8-11eb-8083-9403cab0112f.png)
 **And my result:( It looks not smooth and filled with noise)**
![image](https://user-images.githubusercontent.com/17669514/99941242-a8855c80-2da8-11eb-80d1-a97f96e30a3b.png)

**I wonder what dose the upper half green part of the result mean?** 

### **but the result normal map is not good as example, would you please help me ?**

**Comment by hlwang1124** [comment]
Hi, @slz929 . Thank you for your attention.
Following your description, I used `000005_10` in the kitti stereo 2015 testing set. The corresponding disparity image is generated by [GANet](https://github.com/feihuzhang/GANet). I also added your code into `run_example.py`. The results are shown as follows.

The disparity image:
![000005_10](https://user-images.githubusercontent.com/25403947/100251015-4039a500-2f79-11eb-8331-e2bf85a0544f.png)
The normal result:
![normal](https://user-images.githubusercontent.com/25403947/100251071-5182b180-2f79-11eb-8fbd-066961cbebd1.png)
The freespace segmentation result:
![prob_map](https://user-images.githubusercontent.com/25403947/100251212-77a85180-2f79-11eb-92da-9e5ba0744b0d.png)

These results are ok for me. Grateful if you can provide your images to help me further check this problem.
Moreover, the upper half part is green in the normal image because we set the depth of this area (above the camera) as 0. The corresponding code is Line 22 & 23 in `sne_model.py`. It's ok for freespace segmentation, since they are generally below the camera.

**Comment by slz929** [comment]
@hlwang1124   Thanks for your detailed anwser. I like your work.
I re-check my code, My depth is predicted from one CNN  and the other is also the same as yours. I found I made a stupid mistake which I read depth image as rgb image and read rgb image as depth image,  Sorry...
But when I am debugging, I also find it is important to read the depth by RIGHT cv2.imread MODE:
```
depth_image = cv2.imread(depth_path, cv2.IMREAD_UNCHANGED)
```
will get uint16 and single channel disparity image.
Thanks for helping me analyse and solve. Hope your consistent great work.
谢谢，加油
